src/reducers/__tests__/session.ts

```diff
@@ -0,0 +1,387 @@
+    ...action.payload
+  });
+});
+
+test('SET_USER works correctly', () => {
+  const story: Story = {
+    story: 'test story',
+    playStory: true
+  };
+  const payload = {
+    name: 'test student',
+    role: Role.Student,
+    grade: 150,
+    story
+  };
+
+  const action: IAction = {
+    type: SET_USER,
+    payload
+  };
+  const result: ISessionState = reducer(defaultSession, action);
+
+  expect(result).toEqual({
+    ...defaultSession,
+    ...payload
+  });
+});
+
+test('UPDATE_HISTORY_HELPERS works on non-academy location', () => {
+  const payload = '/playground';
+  const historyHelper: HistoryHelper = {
+    lastAcademyLocations: ['/academy/1', '/academy/2'],
+    lastGeneralLocations: ['/academy/1', '/academy/2']
+  };
+
+  const newDefaultSession = {
+    ...defaultSession,
+    historyHelper
+  };
+  const action: IAction = {
+    type: UPDATE_HISTORY_HELPERS,
+    payload
+  };
+  const resultHistory: HistoryHelper = reducer(newDefaultSession, action).historyHelper;
+
+  expect(resultHistory.lastGeneralLocations).toEqual([
+    historyHelper.lastGeneralLocations[1],
+    payload
+  ]);
+  expect(resultHistory.lastAcademyLocations).toEqual(historyHelper.lastAcademyLocations);
+});
+
+test('UPDATE_HISTORY_HELPERS works on academy location', () => {
+  const payload = '/academy/3';
+  const historyHelper: HistoryHelper = {
+    lastAcademyLocations: ['/academy/1', '/academy/2'],
+    lastGeneralLocations: ['/academy/1', '/academy/2']
+  };
+
+  const newDefaultSession: ISessionState = {
+    ...defaultSession,
+    historyHelper
+  };
+  const action: IAction = {
+    type: UPDATE_HISTORY_HELPERS,
+    payload
+  };
+  const resultHistory: HistoryHelper = reducer(newDefaultSession, action).historyHelper;
+
+  expect(resultHistory.lastGeneralLocations).toEqual([
+    historyHelper.lastAcademyLocations[1],
+    payload
+  ]);
+  expect(resultHistory.lastAcademyLocations).toEqual([
+    historyHelper.lastAcademyLocations[1],
+    payload
+  ]);
+});
+
+// Test Data for UPDATE_ASSESSMENT
+const assessmentTest1: IAssessment = {
+  category: 'Mission',
+  globalDeployment: undefined,
+  graderDeployment: undefined,
+  id: 1,
+  longSummary: 'long summary here',
+  missionPDF: 'www.google.com',
+  questions: [],
+  title: 'first assessment'
+};
+
+const assessmentTest2: IAssessment = {
+  category: 'Contest',
+  globalDeployment: undefined,
+  graderDeployment: undefined,
+  id: 1,
+  longSummary: 'another long summary',
+  missionPDF: 'www.comp.nus.edu.sg',
+  questions: [],
+  title: 'updated first assessment'
+};
+
+test('UPDATE_ASSESSMENT works correctly in inserting assessment', () => {
+  const action: IAction = {
+    type: UPDATE_ASSESSMENT,
+    payload: assessmentTest1
+  };
+  const resultMap: Map<number, IAssessment> = reducer(defaultSession, action).assessments;
+
+  expect(resultMap.get(assessmentTest1.id)).toEqual(assessmentTest1);
+});
+
+test('UPDATE_ASSESSMENT works correctly in updating assessment', () => {
```

Comment:
We can add one more test here!

We have not tested whether the old data in the map is retained (we have only checked the latest data we add/update)